### PR TITLE
Standalone Lite: CheckpointIO

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

This PR moves three cluster environments into the standalone package: The base class, the torch io, and xla io checkpoint plugins. The others (hpu, async wrapper) won't be supported in lite for short term. 

**For easy review, please review each commit separately:**

- Move files and update imports
- Deprecate usages in PL

This PR should be merged as rebase+merge.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @tchaton @rohitgr7 @carmocca @justusschock @awaelchli @akihironitta

### CODE REVIEW

**Review by carmocca — APPROVED** [review]
Remember to add the PR number to the separate commits

**Comment by awaelchli** [comment]
Thanks. Included!

**Review by Borda — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Comment by akihironitta** [comment]
@Borda This PR needs "rebase and merge".

File: src/pytorch_lightning/plugins/io/torch_plugin.py [hunk 1]
```diff
@@ -11,86 +11,23 @@
...
 
-        Args:
-            path: Path to checkpoint
-        """
-        fs = get_filesystem(path)
-        if fs.exists(path):
-            fs.rm(path, recursive=True)
-            log.debug(f"Removed checkpoint: {path}")
+    def __new__(cls, *args: Any, **kwargs: Any) -> "TorchCheckpointIO":
+        rank_zero_deprecation(
```
Comment (tchaton) [inline src/pytorch_lightning/plugins/io/torch_plugin.py:29]: @Borda @awaelchli

I thought we agreed we won’t add any warning for all classes moved to `lightning_lite`?

As a user, I don’t mind if the TorchCheckpointIO is coming from lightning_lite. I still want to be able to import it from PyTorch Lightning without warning.

**Review by tchaton — APPROVED** [review]
LGTM !

File: src/pytorch_lightning/plugins/io/torch_plugin.py (hunk 1 above)
Comment (awaelchli) [reply src/pytorch_lightning/plugins/io/torch_plugin.py]: For now, I removed the deprecations. We can add them back later if we feel it is a good time.

**Pull request merged**